doc/source/tempfile.rst

```diff
@@ -0,0 +1,85 @@
+Temporary Files
+===============
+
+Ray will produce some temporary files during running.
+They are useful for logging, debugging & sharing object store with other programs.
+
+Location of Temporary Files
+---------------------------
+
+First we introduce the concept of a session of Ray.
+
+A session contains a set of processes. A session is created by executing
+``ray start`` command or call `ray.init()` in a Python script and ended by
+executiong ``ray stop`` or call `ray.shutdown()`.
+
+For each session, Ray will create a *root temporary directory* to place all its
+temporary files. The path is `/tmp/ray_{ray_pid}@{datetime}` by default.
+
+You are allowed to change the *root temporary directory* in one of these ways:
+
+* Pass ``--temp-dir={your temp path}`` to ``ray start``
+* Specify `temp_dir` when call `ray.init()`
+
+You can also use ``default_worker.py --temp-dir={your temp path}`` to
+start a new worker with given *root temporary directory*.
+
+The *root temporary directory* you specified will be given as it is
+without pids or datetime attached.
+
+Layout of Temporary Files
+-------------------------
+
+A typical layout of temporary files could look like this:
```

Comment:
maybe we should put a raylet example here, since legacy ray is going to be deprecated soon.

Reply:
I have adopted your advice. But `plasma_store_socket_name` is reserved. See the related comment above.